### ISSUE TITLE
mapping p4 to NetASM IR

### ISSUE THREAD

**Issue by hecmay** [NetASM/NetASM-python#8]
Is there any example or tutorial for mapping p4 to NetASM, as advertised in the description?
> NetASM is a network assembler that takes high-level SDN languages (NetKAT, P4, etc.) and maps the primitives to device-specific directives in FPGAs and various chipsets.

I looked through the wiki as well as the course videos, but did not find any useful information for that.

**Comment by msbaz2013** [comment]
No, there isn't any frontend for P4. But, if you are interested, we would love to have it as a contribution!

**Comment by hecmay** [comment]
@msbaz2013 thanks for your reply. Is there any existing frontend there that I can refer to? It seems that examples are just directly composing the instructions like an IR builder.

**Comment by msbaz2013** [comment]
I don't recall if there is any. I started working on one many years ago but then moved on to other projects.
